File: solutions/Java/missing_number_hash_set.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int missingNumber(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums) seen.add(num);
        for (int value = 0; value <= nums.length; value++) {
            if (!seen.contains(value)) return value;
        }
        return -1;
    }
}

File: solutions/Golang/missing_number_hash_set.go
func missingNumber(nums []int) int {
    seen := map[int]bool{}
    for _, num := range nums {
        seen[num] = true
    }
    for value := 0; value <= len(nums); value++ {
        if !seen[value] {
            return value
        }
    }
    return -1
}
